lib/srv/desktop/discovery.go

```diff
@@ -66,7 +66,7 @@ func (s *WindowsService) startDesktopDiscovery(ctx context.Context) error {
 
 		GetCurrentResources: func() types.ResourcesWithLabels { return s.lastDiscoveryResults },
 		GetNewResources:     s.getDesktopsFromLDAP,
-		OnCreate:            s.createDesktop,
+		OnCreate:            s.upsertDesktop,
 		OnUpdate:            s.updateDesktop,
```

Comment:
Should we use upsert for both create and update?

Reply:
I say we leave it as is, since keeping it as an `update` will surface unexpected behavior such as attempted update on a desktop that doesn't exist. AFAIK that scenario can't happen now, and admittedly it probably won't ever happen, but it falls under the same umbrella as my argument about [semantics](https://github.com/gravitational/teleport/pull/9654#discussion_r779119258) (which is actually relevant to this, since the practical outcomes are the same either way.)

Reply:
Works for me!